python/ml4ir/features/feature_layer.py

```diff
@@ -100,7 +102,7 @@ def feature_layer(inputs):
         Reshape ranking features to create features of shape
         [batch, max_num_records, num_features]
         """
-        ranking_features = tf.concat(ranking_features, axis=-1)
+        ranking_features = tf.concat(ranking_features, axis=-1, name="ranking_features")
```

Comment:
what is meant by "ranking features" here?  What other features are there?  Also, are we changing types again?  `ranking_features` starts as a list, now it's a tensor?  I'm surprised mypy lets you do that!

Reply:
There are three two types of features - `ranking_features` and `metadata_features`. The former is set of features that we learn weights on and are required at inference time to get the ranking score. `metadata_features` are needed at training time like `rank`, `organization_id`, `mask`, etc that are not linked to a weight but help us compute custom ranking losses and metrics.